skimage/morphology/misc.py

```diff
@@ -112,16 +105,10 @@ def remove_small_objects(ar, min_size=64, connectivity=1, in_place=False,
     # Raising type error if not int or bool
     _check_dtype_supported(ar)
 
-    if out is not None:
-        in_place = False
-
-    if in_place:
-        out = ar
+    if out is None:
+        out = ar.copy()
     else:
-        if out is None:
-            out = ar.copy()
-        else:
-            out[:] = ar
+        out[:] = ar
```

Comment:
Same here. Please have a look if I reduced this logic correctly.

Reply:
(out is None) would be caught by the first if, I don't understand why this second (if out is None) under else... I would say you fixed a coding error!